nginx_stage/lib/nginx_stage/configuration.rb

```diff
@@ -349,10 +350,16 @@ def set_default_configuration
       self.template_root         = "#{root}/templates"
 
       self.proxy_user       = 'apache'
-      self.nginx_bin        = '/opt/rh/nginx16/root/usr/sbin/nginx'
+      self.nginx_bin        = '/usr/sbin/nginx'
       self.nginx_signals    = %i(stop quit reopen reload)
-      self.mime_types_path  = '/opt/rh/nginx16/root/etc/nginx/mime.types'
-      self.passenger_root   = '/opt/rh/rh-passenger40/root/usr/share/passenger/phusion_passenger/locations.ini'
+      self.mime_types_path  = '/etc/nginx/mime.types'
+
+      if Etc.uname[:release].include? "el6"
```

Comment:
Why not use something like this ? 

```ruby
require 'phusion_passenger'
self.passenger_root = PhusionPassenger.install_spec || PhusionPassenger.infer_install_spec
```

Reply:
This actually does not work if I start the script with a different Ruby than the default. The only reason it is working is cause on webdev08 I manually copied the passenger lib from /usr/lib/ruby/1.8/phusion_passenger/locations.ini to /opt/rh/rh-ruby24/root/usr/lib64/ruby/phusion_passenger/locations.ini. I was doing so because I thought the location of the passenger ruby libdir mattered and was concerned it was in the system Ruby and not the Ruby I wanted to use.

If I add that to nginx_stage now, it will fail:

```
Error -- /opt/rh/rh-ruby24/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- phusion_passenger (LoadError)
	from /opt/rh/rh-ruby24/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /users/PZS0562/efranz/dev/ondemand/nginx_stage/lib/nginx_stage/configuration.rb:3:in `<top (required)>'
	from /users/PZS0562/efranz/dev/ondemand/nginx_stage/lib/nginx_stage.rb:2:in `require_relative'
	from /users/PZS0562/efranz/dev/ondemand/nginx_stage/lib/nginx_stage.rb:2:in `<top (required)>'
	from /opt/rh/rh-ruby24/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /opt/rh/rh-ruby24/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
```

Reply:
See 9d87a6785c6656f43392648499c877e498bba3f0 which appears to work (tested in a VM on CentOS7 and webdev02 RHEL6) though is not exactly your suggestion.

Reply:
I think the current approach of using `passenger-config about root` is ideal as if we ever supported more than EL operating systems the path may change so letting passenger tell us the path is probably most future-proof approach.

Reply:
Also if we ever packaged passenger ourselves in a way that relocated files all we'd have to do is ensure PATH is correct to continue getting correct passenger root, so current approach is good.